Rename Internal Packages

ObliviousHarmony opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

As an alternative to #33195, we should rename all of the `"private": true` packages with an `internal-` prefix. This makes it immediately clear whether a package is private or not at a glance.

### How to test the changes in this Pull Request:

1. Make sure CI passes, since it runs through all of the builds and tests, it should be considered comprehensive.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

ObliviousHarmony committed:
`internal-` Prefixed `@woocommerce/e2e-builds`

This commit adds an internal prefix to the package
in order to make it clear that it's a private package.

ObliviousHarmony committed:
`internal-` Prefixed `@woocommerce/js-tests`

This commit adds an internal prefix to the package
in order to make it clear that it's a private package.
We also removed the changelogger from the package
since it's internal and does not have a changelog.

ObliviousHarmony committed:
`internal-` Prefixed `@woocommerce/notices`

This commit adds an internal prefix to the package
in order to make it clear that it's a private package.
We also removed the changelogger from the package
since it's internal and does not have a changelog.

ObliviousHarmony committed:
`internal-` Prefixed `@woocommerce/style-build`

This commit adds an internal prefix to the package
in order to make it clear that it's a private package.
We also removed the changelogger from the package
since it's internal and does not have a changelog.

ObliviousHarmony committed:
Updated Lock File

ObliviousHarmony committed:
Fixed Notices Build

botwoo commented:
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33218/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Fixed Notices Build` ac8071d54b99b6444d3b3f70928facbd83f9e060
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

ObliviousHarmony committed:
Updated `wc-notices` Script Handle

Since we've changed it to `internal-notices`
we also need to update the handle to match.

roykho commented on packages/js/internal-e2e-builds/project.json:
```diff
@@ -0,0 +1,8 @@
+     "targets": {
+     }
+ }
+ 
```
New line at EOF

roykho commented on packages/js/internal-style-build/project.json:
```diff
@@ -0,0 +1,7 @@
+     "projectType": "library",
+     "targets": {
+     }
+ }
```
New line at EOF

roykho replied:
New line at EOF

roykho reviewed (dismissed):
Just left some minor linting comments. Otherwise tested locally and build works. LGTM!

ObliviousHarmony committed:
JSON Style Fixes
